app/stores/ada/HWVerifyAddressStore.js

```diff
@@ -73,28 +78,68 @@ export default class AddressesStore extends Store {
     if (isLedgerNanoWallet(conceptualWallet)) {
       await this.ledgerVerifyAddress(path, address);
     } else if (isTrezorTWallet(conceptualWallet)) {
-      await this.trezorVerifyAddress(path, address);
+      await this.trezorVerifyAddress(path, address, publicDeriver.getParent().getNetworkInfo());
     } else {
-      throw new Error('AddressStore::_verifyAddress called with unrecognized hardware wallet');
+      throw new Error(`${nameof(HWVerifyAddressStore)}::${nameof(this._verifyAddress)} called with unrecognized hardware wallet`);
     }
 
     this._setActionProcessing(false);
   }
 
-  trezorVerifyAddress: (BIP32Path, string) => Promise<void> = async (
-    path: BIP32Path,
-    address: string
+  trezorVerifyAddress: (BIP32Path, string, $ReadOnly<NetworkRow>) => Promise<void> = async (
+    path,
+    address,
+    network,
   ): Promise<void> => {
+    const config = getCardanoHaskellBaseConfig(network)
+      .reduce((acc, next) => Object.assign(acc, next), {});
+
+    const getAddressParams = (() => {
+      const type = addressToKind(
+        address,
+        'bech32',
+        network
+      );
+      switch (type) {
```

Comment:
TODO: there are some missing address types here. Either a missing feature in Trezor or a missing part of Trezor documentation